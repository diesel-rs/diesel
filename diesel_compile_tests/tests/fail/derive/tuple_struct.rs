#[macro_use]
extern crate diesel;

table! {
    users {
        id -> Integer,
        name -> Text,
        hair_color -> Nullable<Text>,
    }
}

#[derive(AsChangeset)]
//~^ ERROR: all fields of tuple structs must be annotated with `#[diesel(column_name)]`
#[diesel(table_name = users)]
struct User(i32, #[diesel(column_name = name)] String, String);

fn main() {}
