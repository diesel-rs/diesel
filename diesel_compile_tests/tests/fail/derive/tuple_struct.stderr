error: all fields of tuple structs must be annotated with `#[diesel(column_name)]`
  --> tests/fail/derive/tuple_struct.rs:12:10
   |
LL | #[derive(AsChangeset)]
   |          ^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `AsChangeset` (in Nightly builds, run with -Z macro-backtrace for more info)
